en/release-notes/upgrade-guide-v3.0.md

```diff
@@ -54,56 +48,166 @@ Please get the `UITransform` component on the node first, and then use the corre
+
+![image](web-v3.png)
+
+From the above two figures, we can see that the directory generated after building the Web Desktop, Creator v2.4.3 and 3.0 are mostly the same, with the following differences:
+
+1. Creator 3.0 puts all engine related code, such as core modules, physics modules, plugin scripts, etc., into the `web-desktop/cocos-js` directory, which looks clearer than Creator v2.4.3, which was decentralized in the `web-desktop` directory.
+
+    ![image](web-cocosjs.png)
+
+2. Creator v2.4.3 has only one startup script `main.js`, while Creator 3.0 has the following two startup scripts:
+
+    - `index.js` -- Used to do some pre-processing work.
+    - `application.js` -- Used to start the game.
+
+3. The `src/settings.js` used to manage configuration in Creator v2.4.3 is changed to `src/settings.json` in Creator 3.0.
+
+4. The splash screen `splash.png` in Creator v2.4.3 is stored in `settings.json` by default in Creator 3.0.
+
+5. The `style-desktop.css` and `style-mobile.css` in Creator v2.4.3 are combined into a single `style.css` in Creator 3.0.
+
+#### WeChat Mini Game
+
+The directory generated by Creator v2.4.3 after building the WeChat Mini Game is as follows:
+
+![image](wechat-v243.png)
+
+The directory generated by Creator 3.0 after building the WeChat Mini Game is as follows:
+
+![image](wechat-v3.png)
 
-The **compressed texture configuration** is modified to configure the preset in the project settings, then select the image resource in the resource manager, and then select the preset method. After the old project is upgraded, the editor will automatically scan all the compressed texture configurations in the project and sort out several presets. Since it is automatically scanned, the generated name may not be what you want, you can modify it here.
+From the above two figures, we can see that the directory generated after building the WeChat Mini Game, Creator v2.4.3 and 3.0 are mostly the same, with the following differences:
 
-![image](https://user-images.githubusercontent.com/1503156/100603295-d259f880-333f-11eb-8ff9-c985df953d83.png)
+1. Creator 3.0 puts all engine related code, such as core modules, physics modules, plugin scripts, etc., into the `wechatgame/cocos-js` directory. While Creator v2.4.3 scattered part of it in the `wechatgame` directory and part of it in the `wechatgame/cocos` directory.
 
-### TypeScript Reference Tutorial
+    ![image](wechat-cocosjs.png)
 
-- [TypeScript official website](https://www.typescriptlang.org/)
+2. Creator v2.4.3 compiles all the adaptation layer code of mini games into `adapter-min.js`, while Creator 3.0 stores all the adaptation layer code as loose files in the `libs` directory, without compilation.
+
+3. The startup script for Creator v2.4.3 is `main.js`, and for Creator 3.0 it is `application.js`.
+
+4. Creator v2.4.3 records all references of dynamic code in `ccRequire.js`. While Creator 3.0 does not have this feature currently.
```

Comment:
Creator 3.0
-->
**Cocos Creator 3.0**

Creator v2.4.3
-->
**Cocos Creator v2.4.3**

Reply:
While Creator 3.0 does not have this feature currently.
-->
While __Cocos Creator v3.0__ currently does not have this feature.